lib/src/extensions.dart

```diff
@@ -165,17 +165,24 @@ extension MinutesExtension on MinuteSlotSize {
   }
 }
 
-extension MyList on List<CalendarEventData> {
+extension MyList<T extends Object?> on List<CalendarEventData<T>> {
   // Below function will add the new event in sorted manner(startTimeWise) in
   // the existing list of CalendarEventData.
 
-  void addEventInSortedManner(CalendarEventData event) {
+  void addEventInSortedManner(
+    CalendarEventData<T> event, [
+    EventSorter<T>? sorter,
+  ]) {
     var addIndex = -1;
 
     for (var i = 0; i < this.length; i++) {
-      if ((event.startTime?.getTotalMinutes ?? 0) -
-              (this[i].startTime?.getTotalMinutes ?? 0) <=
-          0) {
+      var result = sorter?.call(event, this[i]) ?? 0;
```

Comment:
can we do like this, and remove the `result == 0` condition. Also, use generics with `_startTimeWiseCompare` function.
```dart
var result = (sorter ?? _startTimeWiseCompare).call(event, this[i]);
```